File: app/src/main/java/org/wen/gank/tools/AppDatabase.java
package org.wen.gank.tools;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import org.wen.gank.model.GankModel;

@Database(entities = {GankModel.class}, version = AppDatabase.version)
@TypeConverters({Converters.class})
public abstract class AppDatabase extends RoomDatabase {

    public static final int version = 1;

    public abstract GankDao gankDao();
}

File: app/src/main/java/org/wen/gank/tools/AppDatabase.kt
package org.wen.gank.tools

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

import org.wen.gank.model.GankModel

@Database(entities = [GankModel::class], version = AppDatabase.version)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {

  abstract fun gankDao(): GankDao

  companion object {
    const val version = 1
  }
}
